app/assets/javascripts/postScribe.js

```diff
@@ -0,0 +1,23 @@
+var waitingOnPostscribe = setInterval(function () {
+
+  if (typeof postscribe === "undefined") {
+    return false;
+  }
+
+  clearInterval(waitingOnPostscribe);
+  var els = document.getElementsByClassName("ltag_gist-liquid-tag");
+  for (var i = 0; i < els.length; i++) {
+    let current = els[i];
+    // eslint-disable-next-line no-undef
+    postscribe(current, current.firstElementChild.outerHTML, {
+      beforeWrite: function (context) {
+        return function (text) {
+          if (context.childElementCount > 3) {
+            return "";
+          }
+          return text;
+        }
+      }(current)
+    });
+  }
+}, 500);
```

Comment:
tiny this is missing the end of line

Reply:
Refactor (blocking): Use a [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports).

We already have the [postscribe](https://www.npmjs.com/package/postscribe) npm package installed as it's used by the post editor. Since this code waits for the postscribe function to exist via a setInterval, I think it makes more sense to remove the entire chunk of of code including the `<% if @article.processed_html.include? "ltag_gist-liquid-tag" %>` in show.html.erg and promote the JS that's current in this new file, to the article [pack file](https://docs.forem.com/frontend/webpacker/) and do a dynamic import of the postscribe package instead in there.

https://github.com/forem/forem/blob/a1ec0be3ce6300e6f5bd7e46f6de04c521cc8752/app/javascript/packs/articlePage.jsx#L1-L67

You can do something like this at the end of the app/javascript/packs/articlePage.jsx file

```javascript
const gistTags = document.querySelectorAll('.ltag_gist-liquid-tag');

// Only load scripts for gists if needed
if (gistTags.length > 0) {
  import('postscribe').then(({ default: postscribe }) => {
    for (const gistTag of gistTags) {
      postscribe(gistTag, gistTag.firstElementChild.outerHTML, {
        beforeWrite(context) {
          return function (text) {
            if (context.childElementCount > 3) {
              return '';
            }

            return text;
          };
        },
      });
    }
  });
}
```